server/broadcast.go

```diff
@@ -192,7 +197,7 @@ func selectOrchestrator(n *core.LivepeerNode, params *streamParameters, cpl core
 
 	for _, tinfo := range tinfos {
 		var sessionID string
-		var balance *core.Balance
+		var balance Balance
```

Comment:
For anyone that is wondering why this type declaration changed: when `balance` was declared as the concrete type `*core.Balance` its nil value is different than the nil value for the `Balance` interface. As a result, even if `balance` was nil, you would get false when comparing it to the explicit `nil`. See https://golang.org/doc/faq#nil_error for more details!

Reply:
This is only true because we're also using the interface on the `BroadcastSession` type definition right?

Reply:
Yep so if we declared `balance` as `*core.Balance` then the comparison `b.Balance != nil` would evaluate to true if `balance` is nil when we set `b.Balance = balance`